src/Workspaces/Core/Portable/Remote/DefaultRemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal partial class DefaultRemoteHostClientServiceFactory
+    {
+        public class RemoteHostClientService : IRemoteHostClientService
+        {
+            private readonly Task<RemoteHostClient> _instanceTask;
+
+            public RemoteHostClientService(Workspace workspace)
+            {
+                var remoteHostClientFactory = workspace.Services.GetService<IRemoteHostClientFactory>();
+                if (remoteHostClientFactory == null)
+                {
+                    // no implementation of remote host client
+                    return;
+                }
+
+                _instanceTask = Task.Run(() => remoteHostClientFactory.CreateAsync(workspace, CancellationToken.None), CancellationToken.None);
```

Comment:
> well, I think that is making things that should be just simple unnecessarily more complex. 

The code above is complex because now you've kicked off potentially expensive work in a non-cancellable fashion.  We should avoid that when possible.  In this case, it seems trivial to avoid it as your method below actually has a cancellation token that can be used :)

> and use LazyAsync to make the creation happen only once. which I am not sure what it is buying us.

It means that things are actually cancellable instead of kicking off work that can't be canceled.  We thread the cancellation tokens through the system for a reason.  It's so clients can say "stop what you're doing, i don't need that result anymore".  In this PR the code doesn't follow that pattern.  You kick off work that can't be canceled, just to get a single value, despite exposing the result of that work through an API that accepts cancellation.

We already have a pattern for that: AsyncLazy.  It allows us to ensure we only have a single value that is created, and it allows for us to stop trying to create it if the caller doesn't need it.